pkg/generate/app/cmd/newapp.go

```diff
@@ -1113,6 +1112,25 @@ func (c *AppConfig) run(acceptors app.Acceptors) (*AppResult, error) {
 	}, nil
 }
 
+// checkCircularReferences ensures there are no builds that can trigger themselves
+// due to an imagechangetrigger that matches the output destination of the image.
```

Comment:
i don't see how that documentation adds value nor it is it the kind of thing we tend to document anywhere else.  It's clear from the type what objects is, and we don't document it anywhere else that we take an app.Objects parameter in a function (AddRoutes, AddServices).  Not to mention it's a private function so it's not even like it's an api we expect people to consume.  The only use is within this package.

And "might contain a buildconfig" is implied by the fact that the function says it's checking the builds, and in the future it might expand to check other things (non-buildconfig related) anyway.

I will add the comment in this case since you've requested it and i'm updating the file anyway, but in general this doesn't seem like a valuable change to make people iterate on.